### PR TITLE
Feature/anaconda setup

### PR DESCRIPTION
**Pull request by iparask**
I removed commands module. The change from PR #873 worked like a charm. I believe that the rest are a good starting point. It works in localhost when Anaconda is present.

### CODE REVIEW

**Comment by marksantcroos** [comment]
Thanks!
Given that there is a switch now, why can't the switch be used to switch within the bootstrapper, instead of duplicating 95% of the code?

**Comment by andre-merzky** [comment]
I think it was my  suggestion to create a new bootstrapper.  I guess at the end it will depends on how similar they'll be in terms of code: if the current bootstrapper ends up with `if/then/else` all over the place I'd prefer a new one.  If the diff at the end is minimal, then I see no problem in having one universal one.

My impression was that anaconda differs fundamentally in how virtualenvs are created / managed, and how modules are installed.  Given that the bootstrapper code is mostly about exactly that, I would think that a new bootstrapper would make sense.  Maybe my assumptions are flawed?

File: src/radical/pilot/bootstrapper/default_bootstrapper_anaconda.sh [hunk 1]
```diff
@@ -0,0 +1,1459 @@
...
+#!/bin/sh
+
+# some inspection for logging
+hostname
+
+# make sure we use the correct sandbox
+cd $SANDBOX
+
+# activate virtenv
+source activate $VIRTENV
```
Comment (andre-merzky) [inline src/radical/pilot/bootstrapper/default_bootstrapper_anaconda.sh]: How can that possibly work?  Does an `activate` script exist in `$SANDBOX`?  Where does it come from?

**Comment by marksantcroos** [comment]
Ok, I'll take that back, its 92% instead of 95%.

```
$ wc -l default_bootstrapper.sh 
    1459 default_bootstrapper.sh
$ wc -l others 
    1331 others
```

File: src/radical/pilot/bootstrapper/default_bootstrapper_anaconda.sh (hunk 1 above)
Comment (iparask) [inline src/radical/pilot/bootstrapper/default_bootstrapper_anaconda.sh]: That is how it is done! I do not know more yet.

**Comment by iparask** [comment]
That 3% made the difference!! :P 

I agree it can be one bootstrapper. But I am not sure how it is should be done....

And we need something stable for the next tutorial

File: src/radical/pilot/bootstrapper/default_bootstrapper_anaconda.sh (hunk 1 above)
Comment (andre-merzky) [inline src/radical/pilot/bootstrapper/default_bootstrapper_anaconda.sh]: I don't get it.  `source` is a shell command which does not respect `$PATH` or anything else for its arguments.  So `activate` needs to be in the dir the `source` command is called in.  That seems to be `$SANDBOX` per 3 lines above.  How did the `activate` command get there?

Comment (iparask) [inline src/radical/pilot/bootstrapper/default_bootstrapper_anaconda.sh]: So there is an `activate` script in the path of the virtual env. `$VIRTENV` has the whole path pointing to the environment. It really does no matter from where you run the commands. I can paste the activate script if you want.

I assume that source looks at the VE path for activate and runs it.

Comment (andre-merzky) [inline src/radical/pilot/bootstrapper/default_bootstrapper_anaconda.sh]: I trust you that it works -- it is just that I am puzzled...  I was wrong BTW wrt. the `PATH` interpretation: source *does* search in `$PATH`.  But I don't see where `$PATH` is set to include the virtenv -- that looks like a chicken and egg problem to me... :P  

Anyway, if it works, it works, no worries!

Thanks, Andre.

**Comment by andre-merzky** [comment]
> I am not sure how it is should be done....

You should leave the config entry in place -- but the PilotLauncher component needs to forward that information to the bootstrapper then.  For that you likely want to introduce a new command line flag, based on that you'd set a variable in the bootstrapper, and then `if/then/else/fi` on all relevant places...

**Comment by iparask** [comment]
Ah! Okay I will give it a try!

**Comment by andre-merzky** [comment]
Hey Iannis -- I made a  couple of small  changes, please check if that still works as expected with Anaconda.  There is one semantic change: `bootstrap_2.sh` now   uses `/bin/bash` instead of `/bin/bash -l` -- was there a specific reason to use a login shell at that point?

**Comment by marksantcroos** [comment]
Let me give this a try on SuperMUC.

**Comment by andre-merzky** [comment]
ack, thanks!

**Comment by iparask** [comment]
As far as I am concerned ```-l``` does not matter. It still works

**Comment by andre-merzky** [comment]
> As far as I am concerned -l does not matter. It still works

Ok, good.  In general, we try to stay away from `-l`, as it pulls in user specific settings which make it hard to reproduce the environment on problems...

**Comment by andre-merzky** [comment]
Mark, can you give it a run on supermuc, please?  Thanks!

File: src/radical/pilot/bootstrapper/default_bootstrapper.sh [hunk 2]
```diff
@@ -1210,6 +1229,7 @@ mkdir -p "$VIRTENV"
 echo "VIRTENV : $VIRTENV"
 VIRTENV=`(cd $VIRTENV; pwd -P)`
 echo "VIRTENV : $VIRTENV (normalized)"
+rmdir "$VIRTENV" 2>/dev/null
```
Comment (marksantcroos) [inline src/radical/pilot/bootstrapper/default_bootstrapper.sh]: Whats the rationale for this? To remove it if it is empty?

Comment (andre-merzky) [inline src/radical/pilot/bootstrapper/default_bootstrapper.sh]: It's complicated :P

pip/python do not play together well when one is using a symlinked path, and the other isn't.  So we try to get the normalized path for the virtenv location (`pwd -P`).  If the virtenv exists, that is not a problem - if it does not, it is.  So we create the dir (`mkdir -p` will not complain if it exists), do our check, and remove it again.  The removal will fail on a non-empty dir, ie. on an existing virtenv, so we redirect the error.

An alternative approach would be to do an `if test -d` ...

**Comment by andre-merzky** [comment]
Thanks Iannis, this seems to be working as expected so far! :D

**Pull request merged**